kernel/include/arch/x86/drivers/keyboard/ps2.h

```diff
@@ -1,14 +1,27 @@
 #include <asm/asm_cpp.h>
+#include <cstdint>
 #include <klog/klog.h>
-std::pair<uint8_t, bool> decode_keyboard_scancode(uint8_t scancode, uint8_t translation_table[], constexpr uint8_t FLAG_RELEASED)
+#include <utility>
+std::pair<std::uint8_t, bool> decode_keyboard_scancode(std::uint8_t scancode, const std::uint8_t* translation_table, const std::size_t table_size,
+                                                       const std::uint8_t FLAG_RELEASED)
 {
-  uint8_t decoded_keypress = 0;
-  bool is_pressed = true; // false means released
-  
-  if (scancode &(!FLAG_RELEASED)< (sizeof(translation_table) / sizeof(translation_table[0])))
-  {
-      decoded_keypress = translation_table[scancode];
-  }
-    is_pressed =scancode&FLAG_RELEASED;
-  return std::make_pair(scancode,is_pressed);
+    std::uint8_t decoded_keypress = 0;
+    bool is_pressed = !(scancode & FLAG_RELEASED); // false means released
+    // klog::log("SCANCODE:: %2X, %2X", scancode, scancode & (~FLAG_RELEASED));
+    // klog::log("SIZE:: %d", table_size);
+    if ((scancode & (~FLAG_RELEASED)) < table_size)
```

Comment:
probably refactor this into an actual variable, something like "actual_scancode"